confluent_platform/datadog_checks/confluent_platform/data/metrics.yaml

```diff
@@ -0,0 +1,1276 @@
+        - kafka.server:type=DelayedOperationPurgatory,name=PurgatorySize,delayedOperation=Fetch
+        # Number of requests waiting in the producer purgatory. This should be non-zero when acks=all is used on the producer.
+        - kafka.server:type=DelayedOperationPurgatory,name=PurgatorySize,delayedOperation=Produce
+        # Maximum lag in messages between the follower and leader replicas. This is controlled by the `replica.lag.max.messages` config.
+        - kafka.server:type=ReplicaFetcherManager,name=MaxLag,clientId=Replica
+        # Number of leaders on this broker. This should be mostly even across all brokers. If not, set auto.leader.rebalance.enable to true on all brokers in the cluster.
+        - kafka.server:type=ReplicaManager,name=LeaderCount
+        # Number of partitions on this broker. This should be mostly even across all brokers.
+        - kafka.server:type=ReplicaManager,name=PartitionCount
+        # Number of partitions whose in-sync replicas count is less than minIsr.
+        - kafka.server:type=ReplicaManager,name=UnderMinIsrPartitionCount
+        # Number of under-replicated partitions (| ISR | < | all replicas |). Alert if value is greater than 0.
+        - kafka.server:type=ReplicaManager,name=UnderReplicatedPartitions
+      attribute:
+        Value:
+          alias: $domain.$type.$name
+          metric_type: gauge
+
+  # Kafka Broker Controller Metrics (Yammer Metric Gauge)
+  - include:
+      domain: kafka.controller
+      bean:
+        # Number of partitions that don’t have an active leader and are hence not writable or readable. Alert if value is greater than 0.
+        - kafka.controller:type=KafkaController,name=OfflinePartitionsCount
+        # Number of active controllers in the cluster. Alert if the aggregated sum across all brokers in the cluster is anything other than 1 because there should be exactly one controller per cluster.
+        - kafka.controller:type=KafkaController,name=ActiveControllerCount
+      attribute:
+        Value:
+          alias: $domain.$name
+          metric_type: gauge
+
+  # Kafka Broker Server Metrics Regex (Yammer Metric Gauge)
+  - include:
+      domain: kafka.server
+      bean_regex:
+        # Lag in number of messages per follower replica. This is useful to know if the replica is slow or has stopped replicating from the leader.
+        - kafka\.server:type=FetcherLagMetrics,name=ConsumerLag,.*
+      attribute:
+        Value:
+          alias: $domain.fetcher_lag.$name
+          metric_type: gauge
+
+  # Kafka Broker Cluster Metrics Regex (Yammer Metric Gauge)
+  - include:
+      domain: kafka.cluster
+      bean_regex:
+        # Number of partitions whose in-sync replicas count is less than minIsr. These partitions will be unavailable to producers who use acks=all.
+        - kafka\.cluster:type=Partition,name=UnderMinIsr,.*
```

Comment:
Not sure how to read this metric from confluent docs, there is both `topic` and `partition` in the bean. How come the metric can represent a number of partition?

Also can you explain what aggregation is being made? I'm afraid it'll submit multiple beans under the same metric name and only keep the latest value because it's a gauge.

Reply:
By default all key/value pairs from the bean name are used as tags. 

Here is an example of the bean that will be matched by this rule `kafka.cluster:type=Partition,name=UnderMinIsr,topic=_confluent-ksql-default__command_topic,partition=0`

And this is the resulting metric from `agent check` command:

```
    }, {
      "metric" : "kafka.cluster.partition.under_min_isr",
      "host" : "default",
      "interval" : 0,
      "source_type_name" : "JMX",
      "type" : "gauge",
      "points" : [ [ 1584357329, 0.0 ] ],
      "tags" : [ "instance:confluent_platform-localhost-31001", "jmx_domain:kafka.cluster", "name:UnderMinIsr", "topic:_confluent-ksql-default__command_topic", "partition:0", "type:Partition" ]
    }, {
```